### ISSUE TITLE
Definitions from foreign protocols are not ignored on Xcode 12

### ISSUE THREAD

**Issue by stjernegard** [peripheryapp/periphery#108]
After upgrading to Xcode 12, I am having an issue that Periphery reports a lot of false positives for things that will be called by UIKit.

I've reproduced the issue with a completely blank iOS App template. After creating the template in Xcode, here's the full list of commands I make to get the issue:

```
$ xcodebuild -scheme TestApp clean build > build.log
$ sed -i '' 's/\\=checked/=checked/g' build.log
$ periphery scan --use-build-log build.log --project "TestApp.xcodeproj"  --schemes "TestApp" --targets "TestApp" --retain-public
```

Periphery then gives me the output

```
TestApp/TestApp/AppDelegate.swift:11:7: warning: Class 'AppDelegate' is unused
TestApp/TestApp/AppDelegate.swift:15:22: warning: Parameter 'application' is unused
TestApp/TestApp/AppDelegate.swift:15:52: warning: Parameter 'launchOptions' is unused
TestApp/TestApp/AppDelegate.swift:22:119: warning: Parameter 'options' is unused
TestApp/TestApp/AppDelegate.swift:22:22: warning: Parameter 'application' is unused
TestApp/TestApp/AppDelegate.swift:28:22: warning: Parameter 'application' is unused
TestApp/TestApp/AppDelegate.swift:28:52: warning: Parameter 'sceneSessions' is unused
TestApp/TestApp/SceneDelegate.swift:10:7: warning: Class 'SceneDelegate' is unused
TestApp/TestApp/ViewController.swift:12:19: warning: Function 'viewDidLoad()' is unused
```

The Sed line is a workaround for the issue reported in #107. That's possibly relevant, but I don't know.

**Comment by SeanMcTex** [comment]
Seeing similar things here for a number of parameters that are required for delegate conformance but that we don't actually use (and which we're correctly indicating are unusued with an underscore). E.g.:

```
    @IBAction func tapGestureRecognized(_ sender: UITapGestureRecognizer) {
        view.endEditing(true)
    }
```

…results in…

`ViewController.swift:140:41: warning: Parameter 'sender' is unused`

**Comment by ileitch** [comment]
@stjernegard I'm unable to reproduce this with v1.8.0, can you please confirm?

@SeanMcTex I've created a new issue for your problem: #131

**Comment by ileitch** [comment]
@stjernegard is your AppDelegate annotated with `@main`?

**Comment by stjernegard** [comment]
`@UIApplicationMain` yes, but it's actually fixed with 1.8.0, so I think this issue can be closed. Thanks for looking into it :)
